internal/provider/network_connection_resource.go

```diff
@@ -0,0 +1,172 @@
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createReq := &network.CreateNetworkConnectionRequest{
+		NetworkId:   data.NetworkID.ValueString(),
+		Params:      nil,
+		Description: data.Description.ValueString(),
+	}
+
+	switch {
+	case data.AWS != nil:
+		aws := &network.CreateAWSNetworkConnectionRequest{}
+		createReq.Params = &network.CreateNetworkConnectionRequest_Aws{
+			Aws: aws,
+		}
+		switch {
+		case data.AWS.Peering != nil:
+			aws.Type = &network.CreateAWSNetworkConnectionRequest_Peering{
+				Peering: &network.CreateAWSNetworkConnectionPeeringRequest{
+					VpcId:         data.AWS.Peering.VPCID.ValueString(),
+					AccountId:     data.AWS.Peering.AccountID.ValueString(),
+					RegionId:      data.AWS.Peering.RegionID.ValueString(),
+					Ipv4CidrBlock: data.AWS.Peering.IPv4CIDRBlock.ValueString(),
+					Ipv6CidrBlock: data.AWS.Peering.IPv6CIDRBlock.ValueString(),
+				},
+			}
+		default:
+			resp.Diagnostics.AddError("misconfiguration", "\"aws.peering\" must be specified")
+			return
+		}
+	case data.Google != nil:
+		createReq.Params = &network.CreateNetworkConnectionRequest_Google{
+			Google: &network.CreateGoogleNetworkConnectionRequest{
+				Name:           data.Google.Name.ValueString(),
+				PeerNetworkUrl: data.Google.PeerNetworkURL.ValueString(),
+			},
+		}
+	default:
+		resp.Diagnostics.AddError("misconfiguration", "at least one of \"aws\" or \"google\" must be specified")
+		return
+	}
+
+	opObj, err := r.networkConnectionService.Create(ctx, createReq)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create", err.Error())
+		return
+	}
+
+	op, err := r.sdk.WrapOperation(opObj, err)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create", err.Error())
+	}
+
+	if err = op.Wait(ctx); err != nil {
+		resp.Diagnostics.AddError("failed to create", err.Error())
+	}
+
+	data.ID = types.StringValue(op.ResourceId())
+
+	if !getNetworkConnection(ctx, r.networkConnectionService, op.ResourceId(), data, resp.Diagnostics) {
+		return
+	}
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *NetworkConnectionResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data *NetworkConnectionModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !getNetworkConnection(ctx, r.networkConnectionService, data.ID.ValueString(), data, resp.Diagnostics) {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *NetworkConnectionResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError("Failed to update network connection", "network connections doesn't support updates")
```

Comment:
One of two:
 * network connection~s~ doesn't support updates
 * network connections do~es~n't support updates

Reply:
:D blind copy-paste)